src/power_grid_model/utils.py

```diff
@@ -301,3 +304,56 @@ def _compatibility_deprecated_import_json_data(json_file: Path, data_type: Datas
         raise PowerGridSerializationError("An internal error occured during deserialization")
 
     return result
+
+
+def self_test():
+    """
+    Perform a self-test of the Power Grid Model functionality to check if the installation was successful
+    and there are no build errors, segfaults, undefined symbols, etc.
+
+
+    Raises:
+        AssertionError: If the output data is empty or if the calculation results gives garbage value.
+
+    Notes:
+        This function is designed to validate the basic functionality of data
+        serialization, model instantiation, power flow calculation, and serialization
+        of results using the Power Grid Model library.
+
+    """
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Create a simple JSON input data file in the temporary directory
+        input_data = {
+            "version": "1.0",
+            "type": "input",
+            "is_batch": False,
+            "attributes": {},
+            "data": {
+                "node": [{"id": 1, "u_rated": 10000}],
+                "source": [{"id": 2, "node": 1, "u_ref": 1, "sk": 1e20}],
+                "sym_load": [{"id": 3, "node": 1, "status": 1, "type": 0, "p_specified": 0, "q_specified": 0}],
+            },
+        }
+
+        input_file_path = os.path.join(temp_dir, "input_data.json")
+        with open(input_file_path, "w", encoding="utf-8") as input_file:
+            json.dump(input_data, input_file)
```

Comment:
maybe you can use `pathlib.Path.write_text` instead

```suggestion
        input_file_path = Pathlib(temp_dir) / "input_data.json"
        input_file_path.write_text(json.dumps(input_data))
```